zap/src/main/java/org/zaproxy/zap/extension/api/ApiResponseSet.java

```diff
@@ -63,9 +63,9 @@ public JSON toJSON() {
     @Override
     public void toXML(Document doc, Element parent) {
         parent.setAttribute("type", "set");
-
         for (Entry<String, T> val : values.entrySet()) {
-            Element el = doc.createElement(val.getKey());
+            String sanitizedKey = val.getKey().replace(" ", "-");
+            Element el = doc.createElement(sanitizedKey);
```

Comment:
@thc202 @psiinon is there a way to check and do this only for XML as it might break people's existing use of the JSON responses from the API?